src/v/storage/parser.cc

```diff
@@ -217,22 +217,22 @@ static ss::future<result<model::record_batch_header>> read_header_impl(
 
 ss::future<result<model::record_batch_header>>
 continuous_batch_parser::read_header() {
-    return read_header_impl(get_stream(), *_consumer, _recovery);
+    auto& st = get_stream();
```

Comment:
nit: this one was arguably easier to read before: is this perhaps a place that you coroutinized to add some debug, but it's gone now?

Reply:
Yep, this is leftover from some debugging. I'll actually move the coroutinizing into a separate PR, to keep this one focused on actual fixes.